modules/k8s/tunnel.go

```diff
@@ -0,0 +1,225 @@
+package k8s
+
+// The following code is a fork of the Helm client. The main differences are:
+// - Support testing context for better logging
+// - Support resources other than pods
+// See: https://github.com/helm/helm/blob/master/pkg/kube/tunnel.go
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/transport/spdy"
+	"k8s.io/kubernetes/pkg/kubectl/generate"
+
+	"github.com/gruntwork-io/terratest/modules/logger"
+)
+
+// KubeResourceType is an enum representing known resource types that can support port forwarding
+type KubeResourceType int
+
+const (
+	ResourceTypePod KubeResourceType = iota
+	ResourceTypeService
+)
+
+func (resourceType KubeResourceType) String() string {
+	switch resourceType {
+	case ResourceTypePod:
+		return "pod"
+	case ResourceTypeService:
+		return "svc"
+	default:
+		// This should not happen
+		return ""
+	}
+}
+
+// Tunnel is the main struct that configures and manages port forwading tunnels to Kubernetes resources.
+type Tunnel struct {
+	KubectlOptions *KubectlOptions
+	LocalPort      int
+	RemotePort     int
+	ResourceType   KubeResourceType
+	ResourceName   string
+	Out            io.Writer
+	StopChan       chan struct{}
+	ReadyChan      chan struct{}
```

Comment:
Are all these fields are used outside of this package (esp the channels)? If not, you can make them lowercase to make them private.

Reply:
Ah ok. I didn't know that convention. Will update!